src/hotspot/share/oops/instanceKlass.cpp

```diff
@@ -1938,12 +1936,12 @@ void InstanceKlass::print_nonstatic_fields(FieldClosure* cl) {
   ResourceMark rm;
   fieldDescriptor fd;
   // In DebugInfo nonstatic fields are sorted by offset.
-  GrowableArray<Pair<int,int> > fields_sorted;
+  GrowableArray<Tuple<int, int, FieldInfo> > fields_sorted;
   int i = 0;
   for (AllFieldStream fs(this); !fs.done(); fs.next()) {
     if (!fs.access_flags().is_static()) {
       fd = fs.field_descriptor();
-      Pair<int,int> f(fs.offset(), fs.index());
+      Tuple<int, int, FieldInfo> f(fs.offset(), fs.index(), fs.to_FieldInfo());
```

Comment:
`FieldInfo` contains the `offset`, so that's not necessary. Besides, the `index` is now only used for an `assert` in the `reinitialize` code. Why not get rid of the index argument and the assert?

In other words: Get rid of the `Tuple` changes and have `fields_sorted` be a `GrowableArray` of `FiedlInfo` only.

Reply:
The suggestions make sense, I'll apply them. Thanks!